src/core/TikTok.ts

```diff
@@ -611,7 +611,7 @@ export class TikTokScraper extends EventEmitter {
             }
             this.maxCursor = parseInt(maxCursor === undefined ? cursor : maxCursor, 10);
         } catch (error) {
-            throw error.message ? error.message : error;
+            throw error.message ? new Error(error.message) : error;
```

Comment:
> you're saying that error's stack trace's root will be where asynchronous code spawned instead of some line of code in submitScrapingRequest's try?

Yes. The error below is from a request inside tiktok-scraper. One could not tell where/when tiktok-scraper spawned the request, or anything else about the request. All it tells us is that one of possibly many requests timed out and "good luck figuring out the rest". 🤷‍♀️

```haskell
RequestError: Error: ESOCKETTIMEDOUT
    at new RequestError (~/tiktok-scraper/node_modules/request-promise-core/lib/errors.js:14:15)
    at Request.plumbing.callback (~/tiktok-scraper/node_modules/request-promise-core/lib/plumbing.js:87:29)
    at Request.RP$callback [as _callback] (~/tiktok-scraper/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at self.callback (~/tiktok-scraper/node_modules/request/request.js:185:22)
    at Request.emit (node:events:394:28)
    at Request.emit (node:domain:475:12)
    at ClientRequest.<anonymous> (~/tiktok-scraper/node_modules/request/request.js:819:16)
    at Object.onceWrapper (node:events:513:28)
    at ClientRequest.emit (node:events:394:28)
    at ClientRequest.emit (node:domain:475:12)
    at TLSSocket.emitRequestTimeout (node:_http_client:761:9)
    at Object.onceWrapper (node:events:513:28)
    at TLSSocket.emit (node:events:406:35)
    at TLSSocket.emit (node:domain:475:12)
    at TLSSocket.Socket._onTimeout (node:net:486:8)
    at listOnTimeout (node:internal/timers:557:17)
    at processTimers (node:internal/timers:500:7)
```

Reply:
Isn't it saying where on this part?

`at self.callback (~/tiktok-scraper/node_modules/request/request.js:185:22)`

Reply:
>  Then at least you'd know which of the requests failed.

Sorry, I don't understand. How can you differentiate one request from another? Won't all of them point to the same `request/request.js:185`?

If I understand it well, all the requests will throw identical errors as long as they are from the same type - e.g. all ESOCKETTIMEDOUT will throw that exact error trace above - regardless of what is going on in tiktok-scraper.

Reply:
Oh sorry, I was in a rush: I thought that line of the stack trace was pointing at something in tiktok-scraper. You're right.